src/contracts/factory/FundFactory.ts

```diff
@@ -31,6 +80,167 @@ export class FundFactory extends Contract {
+        'engine',
+        'feeManager',
+        'mlnToken',
+        'participation',
+        'policyManager',
+        'priceSource',
+        'registry',
+        'shares',
+        'trading',
+        'vault',
+        'version',
+      ],
+      result,
+    );
+  }
+
+  public beginSetup(from: Address, args: Settings) {
+    const method = 'beginSetup';
+    const methodArgs = [
+      args.name,
+      args.fees,
+      args.feeRates.map(rate => rate.toString()),
+      args.feePeriods.map(period => Performance.toString()),
+      args.exchanges,
+      args.adapters,
+      args.denominationAsset,
+      args.defaultAssets,
+    ];
+
+    const validate = async () => {
+      const hub = await this.getManagersToHubs(from);
+      this.validateComponentNotSet(hub);
+
+      const registryAddress = await this.getRegistry();
+      const registry = new Registry(this.environment, registryAddress);
+      if (!(await registry.isAssetRegistered(args.denominationAsset))) {
+        throw new DenominationAssetNotRegisteredError();
+      }
+    };
+
+    return this.createTransaction({ from, method, methodArgs, validate });
+  }
+
+  public createAccouting(from: Address) {
+    const validate = async () => {
+      this.validateComponentSet(await this.getManagersToHubs(from));
+      this.validateComponentNotSet((await this.getManagersToRoutes(from)).accounting);
+    };
+
+    return this.createTransaction({ from, method: 'createAccouting', validate });
+  }
+
+  public createFeeManager(from: Address) {
+    const validate = async () => {
+      this.validateComponentSet(await this.getManagersToHubs(from));
+      this.validateComponentNotSet((await this.getManagersToRoutes(from)).accounting);
+    };
+
+    return this.createTransaction({ from, method: 'createFeeManager', validate });
+  }
+
+  public createParticipation(from: Address) {
+    const validate = async () => {
+      this.validateComponentSet(await this.getManagersToHubs(from));
+      this.validateComponentNotSet((await this.getManagersToRoutes(from)).participation);
+    };
+
+    return this.createTransaction({ from, method: 'createParticipation', validate });
+  }
+
+  public createPolicyManager(from: Address) {
+    const validate = async () => {
+      this.validateComponentSet(await this.getManagersToHubs(from));
+      this.validateComponentNotSet((await this.getManagersToRoutes(from)).policyManager);
+    };
+
+    return this.createTransaction({ from, method: 'createPolicyManager', validate });
+  }
+
+  public createShares(from: Address) {
+    const validate = async () => {
+      this.validateComponentSet(await this.getManagersToHubs(from));
+      this.validateComponentNotSet((await this.getManagersToRoutes(from)).shares);
+    };
+
+    return this.createTransaction({ from, method: 'createShares', validate });
+  }
+
+  public createTrading(from: Address) {
+    const validate = async () => {
+      this.validateComponentSet(await this.getManagersToHubs(from));
```

Comment:
If you start making utility functions for these validations, you could add one for "validateFundSetupStarted" which would check if `managersToHubs` is set or not. Instead of repeating this everywhere. It's also not correct to name that error `ComponentNotSetError` then. It should then be named `FundSetupNotStartedError` or so.

Reply:
Basically, use the `component set/not set` validation helpers for the spokes, and use a special one to for validating the state (`fund setup has already been started`, etc.)